merlion/models/utils/rolling_window_dataset.py

```diff
@@ -1,46 +1,89 @@
+
+        :param data: time series data in the format of TimeSeries or DatetimeIndex indexed pandas DataFrame
+        :param target_seq_index: The index of the univariate (amongst all
+            univariates in a general multivariate time series) whose value we
+            would like to use for the future labeling. If target_seq_index = None,
+            it implies that all the sequences are required for the future labeling,
+            in this case, future will be rolled along the sequence dimension.
+        :param n_past: number of steps for past
+        :param n_future: number of steps for future. If target_seq_index = None, n_future = 1 by enforcement.
+        :param shuffle: True or False to randomly shuffle the returning window, please note that the
+            time series data itself will never be shuffled
+        :param ts_index: keep original TimeSeries internally for all the slicing, and output TimeSeries.
+            by default, Numpy array will handle the internal data workflow and Numpy array will be the output.
+        :param batch_size: if >= 1: will yield a list of (past, future) with batch_size
+            if None: will give a one-shot dataset for all the rolling windows.
+        """
+        assert isinstance(data, (TimeSeries, pd.DataFrame)), \
+            "RollingWindowDataset expects to receive TimeSeries or pd.DataFrame data "
+        if isinstance(data, TimeSeries):
+            data = data.align()
+            self.dim = data.dim
+        else:
+            assert isinstance(data.index, pd.DatetimeIndex)
+            assert ts_index is False, "Only TimeSeries data support ts_index = True "
+            self.dim = data.shape[1]
+
         self.batch_size = batch_size
         self.ts_index = ts_index
 
+        self.target_seq_index = target_seq_index
+        if self.target_seq_index is None:
+            logger.info(
+                f"target_seq_index is None, therefore, the future data will be rolling along "
+                f"the entire sequence dimension with timestamp increment by 1, i.e, self.n_future = 1."
+                f"This is the rolling strategy for autoregression algorithm where all the future "
+                f"sequences need to be provided as the training prior. "
+                f"If you are not expecting this behavior, please properly set up the target_seq_index "
+                f"as a valid integer. "
+            )
+            # label rolling along the sequence dimension
+            self._label_axis = 1
+            self.n_future = 1
+        else:
+            # label rolling along the time dimension
+            self._label_axis = 0
+            self.n_future = n_future
+
+        self.n_past = n_past
+
         if ts_index:
             self.data_ts = data
-            self.target_ts = data.univariates[data.names[target_seq_index]]
+            self.target_ts = data.univariates[data.names[target_seq_index]] if self._label_axis == 0 else None
```

Comment:
This fails if the time series is a dataframe.

Reply:
`RollingWindowDataset` class will check like this, so it shall not get into this section if it is a dataframe

       ```
 if isinstance(data, TimeSeries):
            data = data.align()
            self.dim = data.dim
        else:
            assert isinstance(data.index, pd.DatetimeIndex)
            assert ts_index is False, "Only TimeSeries data support ts_index = True "
            self.dim = data.shape[1]
```